consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/PostMergeContext.java

```diff
@@ -135,7 +137,9 @@ public boolean isSyncing() {
     return Optional.ofNullable(syncState.get()).map(s -> !s.isInSync()).orElse(Boolean.TRUE)
         // this is necessary for when we do not have a sync target yet, like at startup.
         // not being stopped at ttd implies we are syncing.
-        && !syncState.get().hasReachedTerminalDifficulty().orElse(Boolean.FALSE);
+        && Optional.ofNullable(syncState.get())
+            .map(s -> !s.hasReachedTerminalDifficulty().get())
+            .orElse(Boolean.FALSE);
```

Comment:
Good spot, looks like previous code could have got a NPE

Reply:
Would be good to have a test for this too so it proves you fixed the bug.

Reply:
> not sure about the .orElse(Boolean.FALSE); as we removed ! at the beginning of the line

Negation has moved into the map as far as I can see. Agree, unit test should reveal any bugs in the bug fix :)

Reply:
I wonder if changing the negation shouldn't result in an orElse(True)